docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md

```diff
@@ -64,56 +51,133 @@ use_pkce = true
+The order of operations is as follows:
 
-Furthermore, Grafana will check for the presence of at least one of the teams specified via the `team_ids` configuration option using the [JMESPath](http://jmespath.org/examples.html) specified via the `team_ids_attribute_path` configuration option. The JSON used for the path lookup is the HTTP response obtained from querying the Teams endpoint specified via the `teams_url` configuration option (using `/teams` as a fallback endpoint). The result should be a string array of Grafana Team IDs. Using this setting ensures that only certain teams is allowed to authenticate to Grafana using your OAuth provider.
+1. Evaluate the `login_attribute_path` JMESPath expression against the OAuth ID token.
+1. Evaluate the `login_attribute_path` JMESPath expression against the JSON data obtained from UserInfo endpoint, which is specified via the `api_url` configuration option.
 
-You can limit access to only members of a given group or list of groups by setting the `allowed_groups` option.
+### Display name
 
-### Login
+You can set a user's display name with [JMESPath](http://jmespath.org/examples.html) using the `name_attribute_path` configuration option. It operates the same way as the `login_attribute_path` option.
 
-Customize user login using `login_attribute_path` configuration option. Order of operations is as follows:
+### Groups
 
-1. Grafana evaluates the `login_attribute_path` JMESPath expression against the ID token.
-1. If Grafana finds no value, then Grafana evaluates expression against the JSON data obtained from UserInfo endpoint. The UserInfo endpoint URL is specified in the `api_url` configuration option.
+Group mappings are made using [JMESPath](http://jmespath.org/examples.html) from the `groups_attribute_path` configuration option.
+Grafana determines a user's groups by following the steps below until it finds the groups:
 
-You can customize the attribute name used to extract the ID token from the returned OAuth token with the `id_token_attribute_name` option.
+1. Evaluate the `groups_attribute_path` JMESPath expression against the OAuth ID token.
+1. Evaluate the `groups_attribute_path` JMESPath expression against the JSON data obtained from UserInfo endpoint, which is specified via the `api_url` configuration option.
 
-You can set the user's display name with JMESPath using the `name_attribute_path` configuration option. It operates the same way as the `login_attribute_path` option.
+The result of the JMESPath expression should be a string array of groups.
 
-> **Note:** `name_attribute_path` is available in Grafana 7.4+.
+You can limit access to Grafana to only members of a given group or list of groups by setting the `allowed_groups` option.
 
-### PKCE
+### Teams
+
+OAuth provider teams are extracted using [JMESPath](http://jmespath.org/examples.html) expression from the `team_ids_attribute_path` configuration option.
```

Comment:
I'm not sure why a user needs to know this information. Does this impact how they complete the task?